### ISSUE TITLE
Blitz causes behaviours to be removed from the user model

### ISSUE THREAD

**Issue by lenvanessen** [putyourlightson/craft-blitz#299]
### Describe the bug
Related:
https://github.com/craftcms/cms/issues/7613

We've run into errors where Blitz would call the User model, before the ` User::EVENT_DEFINE_BEHAVIORS` is called by Craft. This causes the behavior not to register properly and this breaks functionality in our system.

Blitz calls the user to check whether or not to cache a Request, here in `CacheRequestService` in the method `getIsCacheableRequest`. I'm not sure where to look for a fix, we've made a workaround for our system right now, by adding the behavior directly to the user model, rather than waiting for the Event to be called. But perhaps Blitz can do something to ensure not breaking this system, given that it's calling the identity so early in the boot process.

**Comment by brandonkelly** [comment]
Not sure if this is feasible in Blitz’ case, but ideally, plugin code that would involve fetching the logged-in user or rendering a template should be held off until `craft\web\Application::EVENT_INIT` is triggered.

```php
public function init()
{
    parent::init();

    \Craft::$app->on(\craft\web\Application::EVENT_INIT, function() {
        // ...
    });
}
```

**Comment by bencroker** [comment]
Thanks @brandonkelly!

I've put all logic that checks if this is a cacheable request into `Application::EVENT_INIT` in https://github.com/putyourlightson/craft-blitz/commit/410b63c4a5678addc3eeea99f95635dc11bf0aac (in a new branch).
https://github.com/putyourlightson/craft-blitz/blob/410b63c4a5678addc3eeea99f95635dc11bf0aac/src/Blitz.php#L218-L275

Can you let me know if this solves your issue @lenvanessen?

**Comment by lenvanessen** [comment]
@bencroker thanks for the fix, works like a charm.

@brandonkelly thanks for joining in on this discussion as well, it shows a level of dedication to the community that is unparalleled by any other cms or project out there

**Comment by bencroker** [comment]
Great, will do some testing and add to the next release.

**Comment by lenvanessen** [comment]
@bencroker just hit another race condition, with the Scout extension. It seems Blitz' Init event is fired before the init event from Scout, and causing the same issue.

**Comment by bencroker** [comment]
Hmm, can you try prepending the event handler be setting the fourth parameter to `false`? Plugins are ordered alphabetically so **B**litz is initialised before **S**cout.

```php
Event::on(Application::class, Application::EVENT_INIT, 'handler', false);
```

Any thoughts on this workaround @brandonkelly? Feels kind of icky.

**Comment by bencroker** [comment]
FYI released in 3.8.0.
